src/Init/Notation.lean

```diff
@@ -201,6 +201,10 @@ macro_rules
     else
       `(%[ $elems,* | List.nil ])
 
+-- TODO: should be `infix:50 " matches " => fun e p => match e with | p => true | _ => false`
+macro:50 e:term:51 " matches " p:term:51 : term =>
+  `(match $e:term with | $p:term => true | _ => false)
```

Comment:
If this construct is really introduced to the language, we should also get rid of a bunch of predicates like `Syntax.isIdent`, right? But with `matches` you still need to spell out all the binders even though you *know* that they will be ignored. Maybe if it's meant to be part of Lean proper, it should do that automatically?